object/slicer/slicer.go

```diff
@@ -237,25 +237,10 @@ func slice(ctx context.Context, ow ObjectWriter, header object.Object, data io.R
 	return rootID, nil
 }
 
-// headerData extract required fields from header, otherwise throw the error.
-func headerData(header object.Object) (cid.ID, user.ID, error) {
+func initPayloadStream(ctx context.Context, ow ObjectWriter, header object.Object, signer user.Signer, opts Options) (*PayloadWriter, error) {
 	containerID := header.GetContainerID()
 	if containerID.IsZero() {
-		return cid.ID{}, user.ID{}, fmt.Errorf("container-id: %w", ErrIncompleteHeader)
-	}
-
-	owner := header.Owner()
-	if owner.IsZero() {
-		return cid.ID{}, user.ID{}, fmt.Errorf("owner: %w", ErrIncompleteHeader)
-	}
-
-	return containerID, owner, nil
-}
-
-func initPayloadStream(ctx context.Context, ow ObjectWriter, header object.Object, signer user.Signer, opts Options) (*PayloadWriter, error) {
-	containerID, owner, err := headerData(header)
-	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("header container-id is empty: %w", ErrIncompleteHeader)
```

Comment:
unrelated, i.e. separate, change to me

also, if we touch it, i'd improve it like this
```suggestion
		return nil, fmt.Errorf("%w: %w", ErrIncompleteHeader, cid.ErrZero)
```

group error first, then particular one. And reuse common error

Reply:
Updated

Reply:
Separated